registration/include/pcl/registration/ndt_omp.h

```diff
@@ -0,0 +1,521 @@
+                 PointCloudSource& trans_cloud,
+                 Eigen::Matrix<double, 6, 1>& p);
+
+  /** \brief Compute individual point contirbutions to hessian of probability function w.r.t. the transformation vector.
+    * \note Equation 6.13 [Magnusson 2009].
+    * \param[in,out] hessian the hessian matrix of the probability function w.r.t. the transformation vector
+    * \param[in] x_trans transformed point minus mean of occupied covariance voxel
+    * \param[in] c_inv covariance of occupied covariance voxel
+    */
+  void
+  updateHessian(Eigen::Matrix<double, 6, 6>& hessian,
+                const Eigen::Matrix<double, 3, 6>& point_gradient,
+                const Eigen::Matrix<double, 18, 6>& point_hessian,
+                const Eigen::Vector3d& x_trans,
+                const Eigen::Matrix3d& c_inv) const;
+
+  /** \brief Compute line search step length and update transform and probability derivatives using More-Thuente method.
+    * \note Search Algorithm [More, Thuente 1994]
+    * \param[in] x initial transformation vector, \f$ x \f$ in Equation 1.3 (Moore, Thuente 1994) and \f$ \vec{p} \f$ in Algorithm 2 [Magnusson 2009]
+    * \param[in] step_dir descent direction, \f$ p \f$ in Equation 1.3 (Moore, Thuente 1994) and \f$ \delta \vec{p} \f$ normalized in Algorithm 2 [Magnusson 2009]
+    * \param[in] step_init initial step length estimate, \f$ \alpha_0 \f$ in Moore-Thuente (1994) and the noramal of \f$ \delta \vec{p} \f$ in Algorithm 2 [Magnusson 2009]
+    * \param[in] step_max maximum step length, \f$ \alpha_max \f$ in Moore-Thuente (1994)
+    * \param[in] step_min minimum step length, \f$ \alpha_min \f$ in Moore-Thuente (1994)
+    * \param[out] score final score function value, \f$ f(x + \alpha p) \f$ in Equation 1.3 (Moore, Thuente 1994) and \f$ score \f$ in Algorithm 2 [Magnusson 2009]
+    * \param[in,out] score_gradient gradient of score function w.r.t. transformation vector, \f$ f'(x + \alpha p) \f$ in Moore-Thuente (1994) and \f$ \vec{g} \f$ in Algorithm 2 [Magnusson 2009]
+    * \param[out] hessian hessian of score function w.r.t. transformation vector, \f$ f''(x + \alpha p) \f$ in Moore-Thuente (1994) and \f$ H \f$ in Algorithm 2 [Magnusson 2009]
+    * \param[in,out] trans_cloud transformed point cloud, \f$ X \f$ transformed by \f$ T(\vec{p},\vec{x}) \f$ in Algorithm 2 [Magnusson 2009]
+    * \return final step length
+    */
+  double
+  computeStepLengthMT(const Eigen::Matrix<double, 6, 1>& x,
+                      Eigen::Matrix<double, 6, 1>& step_dir,
+                      double step_init,
+                      double step_max,
+                      double step_min,
+                      double& score,
+                      Eigen::Matrix<double, 6, 1>& score_gradient,
+                      Eigen::Matrix<double, 6, 6>& hessian,
+                      PointCloudSource& trans_cloud);
+
+  /** \brief Update interval of possible step lengths for More-Thuente method, \f$ I \f$ in More-Thuente (1994)
```

Comment:
While I appreciate the effort in documentation, it'd be better to create a struct, and provide comments in one place, not multiple places.